### PR TITLE
#160056383: user shares article via social

### PR DESCRIPTION
**Pull request by chidioguejiofor**
#### What does this PR do?
- adds feature that allows users to share article on social media
- creates a SocialShare component
- updates tests and ensures that all the tests are passing

#### Description of Task to be completed?
- user should be able to share articles on social platform

#### How should this be manually tested?
- `git pull` this branch
- update environmental variables using the .env-sample as a reference
- npm start to start the server
-  view an article via `/articles/{slug}`
- underneath the page you would see the Share article section

#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- 160056383
#### Screenshots (if appropriate)
screen shot 2018-09-26 at 1 26 18 pm
screen shot 2018-09-26 at 1 26 17 pm
screen shot 2018-09-26 at 1 26 13 pm
screen shot 2018-09-26 at 1 26 04 pm 1
screen shot 2018-09-26 at 1 26 04 pm
screen shot 2018-09-26 at 1 25 54 pm
screen shot 2018-09-26 at 1 25 16 pm
screen shot 2018-09-26 at 1 25 03 pm
screen shot 2018-09-26 at 1 24 53 pm
screen shot 2018-09-26 at 1 24 19 pm
screen shot 2018-09-26 at 1 24 18 pm
screen shot 2018-09-26 at 5 08 49 am
screen shot 2018-09-26 at 1 28 33 pm

#### Questions:
- None

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (megame24) [inline .vscode/launch.json]: What is the purpose of this file?

File: src/actions/loadArticleAction.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import axiosInstance from '../services/requestHandler';
```
Comment (megame24) [inline src/actions/loadArticleAction.js]: Put all your article related actions in a single file, say `articleActions.js`

File: public/styles/partials/variables.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+$text-color: #545454;
+$link-color: #1a0dab;
+$white: #fff;
+
+$breakpoints: (
+  xs: 576px,
+  sm: 768px,
+  md: 992px,
+  lg: 1200px
+);
```
Comment (JCanaks) [inline public/styles/partials/variables.scss]: resolve EOL

File: public/styles/_social-share.scss [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#social-share{
+    text-align: center;
+    margin-top: 10px;
+    padding: 30px 0  ;
+    border-top: 2.5px solid rgba(0,0,0,0.1);
+}
```
Comment (JCanaks) [inline public/styles/_social-share.scss]: Resolve all EOLs

File: src/actions/updateArticleAction.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import axiosInstance from '../services/requestHandler';
```
Comment (JCanaks) [inline src/actions/updateArticleAction.js]: like innocent said you might want to compile the article related actions into one file

File: src/components/BasicUserInfo.jsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```
Comment (JCanaks) [inline src/components/BasicUserInfo.jsx]: Pls remove this component since its not being used

File: src/components/SocialShare.jsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { Button, Header } from 'semantic-ui-react';
+import propTypes from 'prop-types';
+
```
Comment (JCanaks) [inline src/components/SocialShare.jsx:4]: resolve hound violations

File: src/configs/routes.js [hunk 8]
```diff
@@ -111,7 +111,7 @@ export default [
   },
   {
     type: Route,
-    path: '*',
+    path: 'not-found',
```
Comment (JCanaks) [inline src/configs/routes.js]: Why is the catch all route changed to 'not-found'?

File: src/index.jsx [hunk 9]
```diff
@@ -23,7 +23,7 @@ if (localStorage.token) {
 render(
   <Provider store={store}>
     <Router>
-      <Route component={App} />
+      <App />
```
Comment (JCanaks) [inline src/index.jsx]: why was this change made?

File: src/reducers/createArticleReducer.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
+import types from '../actions/actionTypes';
```
Comment (JCanaks) [inline src/reducers/createArticleReducer.js]: You might want to compile all article related reducers to one file

File: src/views/auth/AuthRoute.jsx [hunk 11]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
```
Comment (JCanaks) [inline src/views/auth/AuthRoute.jsx]: This file is no longer being used pls delete

File: tests/components/BasicUserInfo.test.js [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+  let mountedObj;
+  let divContainer;
+
+  describe('the contents that the Component renders', () => {
+    beforeEach(() => {
+      mountedObj = shallow(<BasicUserInfo />);
+      divContainer = mountedObj.find('div');
+    });
+
+    it('sshould render component', () => {
```
Comment (JCanaks) [inline tests/components/BasicUserInfo.test.js]: `'should'` not `sshould`

**Review by JCanaks — APPROVED** [review]

**Review by mppdg — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]

**Pull request merged**